src/reanimated2/js-reanimated/MapperRegistry.js

```diff
@@ -0,0 +1,116 @@
+export default class MapperRegistry {
+  sortedMappers = [];
+  mappers = new Map();
+
+  updatedSinceLastExecute = false;
+
+  constructor(module) {
+    this._module = module;
+  }
+
+  startMapper(mapper) {
+    this.mappers.set(mapper.id, mapper);
+    this.updatedSinceLastExecute = true;
+    return mapper.id;
+  }
+
+  stopMapper(id) {
+    this.mappers.delete(id);
+    this.updatedSinceLastExecute = true;
+  }
+
+  execute() {
+    if (this.updatedSinceLastExecute) {
+      this.updateOrder();
+      this.updatedSinceLastExecute = false;
+    }
+
+    this.sortedMappers.forEach((mapper) => {
+      if (mapper.dirty) {
+        mapper.execute();
+      }
+    });
+  }
+
+  updateOrder() {
+    class Node {
+      mapper = null;
+      children = [];
+
+      constructor(mapper, parents = [], children = []) {
+        this.mapper = mapper;
+        this.children = children;
+      }
+    }
```

Comment:
I think this can be moved away from this method

Reply:
5ee34fd